site/app/models/User.php

```diff
@@ -215,6 +218,35 @@ public function updateTimeZone(string $time_zone): bool {
         return false;
     }
 
+    /**
+     * Get the user's time zone, in 'nice' format.  This simply returns a cleaner 'NOT SET' string when the
+     * user has not set their time zone.
+     *
+     * @throws NullException Found null user time zone, but this should never happen
```

Comment:
Why do we need this null check if it should never happen? What are the cases that are causing this to be null, and can we better protect against those upstream, by making the change above, or just throwing an exception where `$this->time_zone` is set to null?

Reply:
You make a good point here.  I got rid of the null exception, and changed updateTimeZone to be setTimeZone, so now the only ways to set it are in the constructor and with the setter.  It should never end up null unless the user isn't loaded.